### PR TITLE
wip

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dali/daliadmin/daliadmin.cpp [hunk 1]
```diff
@@ -3219,7 +3239,135 @@ void testThorRunningWUs()
...
+            }
+            else
+            {
+                ++existingPersists;
+                varMapIt++;
+            }
+        }
+        catch (IException *e)
+        {
+            VStringBuffer errMsg("Skpping: %s", name);
```
Comment (ghalliday) [inline dali/daliadmin/daliadmin.cpp:3339]: typo: skipping

File: dali/daliadmin/daliadmin.cpp [hunk 2]
```diff
@@ -3219,7 +3239,135 @@ void testThorRunningWUs()
...
+            e->Release();
+        }
+        ++checked;
+
+        if (0 == (checked % reportInterval))
+        {
+            cycle_t perVarCycles = per100Timer.elapsedCycles() / reportInterval;
+
+            per100Timer.reset();
+            PROGLOG("%sChecked: [%u / %u] - deletes: %u, auxDeletes: %u, persists: %u. [Avg ms: %2.2f]", dryrun?"DRYRUN:":"", checked, total, deletes, auxDeletes, existingPersists, static_cast<unsigned>(cycle_to_microsec(perVarCycles))/1000.0);
```
Comment (ghalliday) [inline dali/daliadmin/daliadmin.cpp:3350]: minor: I think the reported average is in seconds, rather than ms (since you're dividing ms by 1000.0)

Comment (ghalliday) [reply dali/daliadmin/daliadmin.cpp:3350]: ignore that.  Confused microseconds and milli seconds

Comment (jakesmith) [reply dali/daliadmin/daliadmin.cpp:3350]: it's microsecs / 1000.0.
So millionths of sec.  / 1000.0 => millisecs ?

File: dali/daliadmin/daliadmin.cpp [hunk 3]
```diff
@@ -3636,6 +3790,21 @@ int main(int argc, char* argv[])
...
+                    {
+                        CHECKPARAMS(0, 2);
+                        bool dryrun = false;
+                        bool reconstruct = true;
+                        for (unsigned i=1; i<params.ordinality(); i++)
+                        {
+                            const char *param = params.item(i);
+                            if (strieq("dryrun", param))
+                                dryrun = true;
+                            else if (strieq("reconstruct", param))
```
Comment (ghalliday) [inline dali/daliadmin/daliadmin.cpp:3803]: now need noreconstruct?

Comment (jakesmith) [reply dali/daliadmin/daliadmin.cpp:3803]: yes, I'll change it to "noreconstruct"

**Pull request closed without merging**